app2/views.py

```diff
@@ -1,25 +1,156 @@
+from django.shortcuts import redirect
+from django.contrib.auth import logout
+from django.core.mail import send_mail
 from django.db.models import Count, Sum, Avg
+from django.template import Context, Template
+from django.views.generic.list import ListView
+from django.views.generic.detail import DetailView
+from django.contrib.auth import authenticate, login
+from django.core.urlresolvers import reverse, resolve
+from django.contrib.auth.models import User, UserManager
+from django.contrib.auth.decorators import login_required
+from django.http import HttpResponse, HttpResponseRedirect
+from .models import Item, DerivedItem, Ingredient, LoginForm, RegistrationForm, CustomUser
 
 # Create your views here.
 
-def f(x,l=[]):
-    for i in range(x):
-        l.append(i*i)
-    print(l) 
-
-def index(request):
-    f(3)
-    # f(3,[3,2,1])
-    f(3)
-
-    # Get published items
-    latest_list = DerivedItem.objects.get_list_via_filter()
-    item_list = Item.objects.select_related().annotate(sum_ings_price = Sum('ings__price'))
-    context = {'latest_list': latest_list, 'item_list' : item_list}
-    return render(request, 'app2/index2.html', context)
-    #return HttpResponse(reverse(request.resolver_match.namespace + ':index', current_app = request.resolver_match.namespace))
\ No newline at end of file
+#class MyView(View):
+#    def get(self, request):
+#        # <view logic>
+#        return HttpResponse('result')
+
+
+class IndexView(View):
+    #@login_required(login_url='login.html')
+    def get(self, request):
+        # Get published items
+        latest_list = DerivedItem.objects.get_list_via_filter()
+        item_list = Item.objects.select_related().annotate(sum_ings_price = Sum('ings__price'))
+        context = {'latest_list': latest_list, 'item_list' : item_list}
+        return render(request, 'app2/index2.html', context)
+
+
+class ProfileView(View):
+    def get(self, request, pk=-1):
+        if (pk > -1):
+            user_pk_int = int(pk)
+        else:
+            user_pk_int = int(request.GET.get('pk', '0'))
+            pk = user_pk_int
+        try:
+            user = User.objects.get(id=user_pk_int)
+        except:
+            context = Context({ 'user_pk' : pk, 'user_name' : 'Not found', 'user_email' : 'Not found' })      
+        else:
+            context = Context({ 'user_pk' : pk, 'user_name' : user.username, 'user_email' : user.email })
+        return render(request, 'app2/profile.html', context)
+
+
+class RegistrationView(View):
+    def _send_email(self, user):
+        content = "127.0.0.1:8080/app2/confirm/" + str(user.confirmation_code) + "/" + user.username
+        send_mail("Account confirmation", content, settings.EMAIL_HOST_USER, [user.email], fail_silently=False)
+
+
+    def _create_new_user(self, username, last_name, email, password):
+        # https://docs.djangoproject.com/en/dev/topics/auth/default/
+        user = CustomUser.objects.create_user(username, email, password)
```

Comment:
create_user(username, email=None, password=None, **extra_fields) EXTRA FIELDS ? (instead one more save) Look at this  email=None, password=None, and compare with you code create_user(username, email, password)